### PR TITLE
feat!: don't load bundles with opt items, add api for loading them

### LINKED ISSUE #59: [Feature] Api for manually loading bundles
See https://github.com/nvim-neorocks/rocks-lazy.nvim/issues/8.

We could automatically skip loading a bundle if all of its plugins are `opt` (technically a breaking change).

### PR DESCRIPTION
**Pull request by mrcjkb**
Closes #59

### CODE REVIEW

File: lua/rocks-config/internal.lua [hunk 1]
```diff
@@ -208,14 +208,119 @@ function rocks_config.configure(rock, config)
...
+                string.format(
+                    "[rocks-config.nvim]: Bundle '%s' has invalid plugin '%s'. Did you make a typo, or is the plugin not installed?",
+                    bundle_name,
+                    nonexistent_bundle_item
+                ),
+                vim.log.levels.ERROR
+            )
+        end)
+        return
+    end
```
Comment (mrcjkb) [inline lua/rocks-config/internal.lua]: question: Should we skip loading the bundle if **all** of the items are `opt` or if **any** of them are?

Comment (mrcjkb) [reply lua/rocks-config/internal.lua]: I think **any** makes the most sense. rocks-config could provide an API to get the remaining plugins in a bundle. Then modules like rocks-dev.nvim could use that to add the plugins to the rtp before sourcing the bundle.

Comment (mrcjkb) [reply lua/rocks-config/internal.lua]: Done.

**Review by vhyrro — APPROVED** [review]
Can't find any issues by looking at the code. Looks good to me :)

**Pull request merged**